accepted/2021/workloads/workload-target-imports.md

```diff
@@ -0,0 +1,120 @@
+- `Directory.Build.targets`
+- (Rest of) .NET SDK targets
+- Old location for Windows, WindowsDesktop, and workload targets imports
+
+#### `Directory.Build.targets` import location
+
+Conceptually, `Directory.Build.targets` is imported after the body of the main project file. The exact location it is imported is not something most developers likely think about, but it is a good place to put common build logic that depends on properties set in the project file, such as the `TargetFramework`.
+
+There's not a perfect place to import `Directory.Build.targets`. However, given what we've learned, it may be that the best place to import it is after the TargetFramework parsing, and before the workloads are imported.  That way the logic in `Directory.Build.targets` would still be able to depend on the parsed TargetFramework information, but would not be able to override all of the targets and properties set by the .NET SDK and MSBuild common targets that it can today.
+
+## Proposed Solution: Change import location of `Directory.Build.targets`
+
+Change`Directory.Build.targets` to be imported after TargetFramework parsing but before workloads and most of the .NET SDK and common targets. This would be a big breaking change but might not affect most people who use `Directory.Build.targets`. It would break people who use `Directory.Build.targets` to override properties or targets in the MSBuild common targets.
+
+#### Pros:
+
+- Logic to enable workloads in `Directory.Build.targets` would “just work”, which is likely what developers expect
+
+#### Cons
+
+- This would be a breaking change, possibly a very big one
+- Implementation would be somewhat complicated:
+  - Logic would be split betweenthe .NET SDK and MSBuild, as the .NET SDK would now be responsible for importing `Directory.Build.targets` for projects that use the .NET SDK, but other projects would still need MSBuild to import `Directory.Build.targets`
+  - The new behavior would need to be behind a change wave and ideally a separate opt-out flag
+- This goes against the original intent behind `Directory.Build.targets`, which was that it would be imported after all built-in targets.  However, the .NET SDK (or any MSBuild SDK) already break this.
+  - As part of this work we could possibly add another extension point that allows you to specify `.targets` files that are imported after all other projects or imports.  This would likely be an MSBuild engine feature.
+
+#### Proposed Breaking Change Mitigations
+
+- An opt out property would be implemented to allow for preserving existing behavior.
+- We would like to make this change only apply to projects targeting .NET 6 and higher.
+
+## Alternative solutions
+
+### Extension Point via Property
+
+Support an `AfterTargetFrameworkInferenceTargets` property. This property could be used by creating a `Directory.AfterTargetFrameworkInference.targets` file and putting the following in the `Directory.Build.props` file located in the same folder:
```

Comment:
If it is to accommodate workloads, why not introduce something like `Workload.Build.props` and `Workload.Build.targets`?